src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs

```diff
@@ -36,6 +38,24 @@ public class SecretManager : IDisposable, ISecretManager
         private string _repositoryClassName;
         private DateTime _lastCacheResetTime;
 
+        internal const string AzureFunctionsSignature = "AzFu";
+
+        // Checksum seed for master keys. The string transformation here results in a
+        // a seed that is versioned in the least significant bits (as a ulong):
+        // BitConverter.ToUInt64(Encoding.ASCII.GetBytes("Master00").Reverse().ToArray());
+        // If there is a need to version, use "Master01" as the next string literal.
+        internal const ulong MasterKeySeed = 0x4d61737465723030;
```

Comment:
>MasterKeySeed [](http://example.com/codeflow?start=29&length=13)

An important question for the team is how to 'bucket' keys by generating each set with a unique seed. We'd typically do this in situations where response/remediation may take a different path. In general, a good strategy is to group all user-facing (customer handled) secrets with one seed, and two organize other generated keys (which are intended to be internal-facing only) with another.

Reply:
In our case, all of these keys are user-facing - the keys just have different authorization levels. So the severity of an issue would vary based on key type. However, in all cases I think the response/remediation would be the same. So I'm not sure if we need the three buckets. @brettsam / @fabiocav what do you think?

Reply:
The bucketing is most helpful for routing problems to the appropriate audience for response. If all keys are user-facing, you could definitely get by with one.

There is arguably a benefit in response, however, in having a more granular categorization, simple because more information in reporting maximizes actionability. i.e., if the distinction between master, system, and function keys in any ways maximizes motivation to fix, lowers costs to get oriented and respond, could be worth it. Certainly the scanning costs aren't high for bucketing.

My personal opinion is that it's a close call but I'd favor having more information/retaining the three buckets. But I wouldn't argue if you'd prefer to keep it simple.

Reply:
@mathewc , removed all deps.json changes (resulted from me baselining a debug build, I believe). will work on getting all functional tests running today.

Reply:
I'm fine with retaining the buckets, thanks.